protocol/Telemetry.proto

```diff
@@ -0,0 +1,200 @@
+// BSD 2-Clause License
+//
+// Copyright (c) 2020 Emily Horsman, Tanner Ryan. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this
+//    list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice,
+//    this list of conditions and the following disclaimer in the documentation
+//    and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+syntax = "proto3";
+
+/**
+ * Telemetry wire protocol.
+ */
+message Telemetry {
+    oneof message {
+        Registration registration = 1;
+        Provisioning provisioning = 2;
+        Payload payload = 3;
+        Request request = 4;
+    }
+}
+
+/**
+ * Registration frame (microcontroller -> backend).
+ *
+ * Conditions:
+ *  - MQTT connection established, re-established
+ *  - Peripheral connected, disconnected
+ */
+message Registration {
+    /**
+     * Firmware version identifier.
+     */
+    uint32 version = 1;
+
+    /**
+     * Microcontroller unique identifier (MAC address).
+     */
+    bytes uuid = 2;
+
+    /**
+     * Microcontroller IP address.
+     */
+    bytes ip = 3;
+
+    /**
+     * List of connected peripherals.
+     */
+    repeated Peripheral_t peripherals = 4;
+    message Peripheral_t {
+        uint32 busId = 1;
+        uint32 busAddr = 2;
+        bytes generalCallResp = 3;
+    }
+}
+
+/**
+ * Provisioning frame (backend -> microcontroller).
+ *
+ * Conditions:
+ *  - Received Registration frame from microcontroller.
+ *  - Received new read definitions from corresponding peripheral controller.
+ */
+message Provisioning {
+    /**
+     * Default bus address for peripheral.
+     */
+    uint32 busAddr = 1;
```

Comment:
Should be `bytes`, I need to update the JSON sent from the peripheral controller to reflect this